File: perm_check/perm_check.py
# https://app.codility.com/demo/results/trainingWP6RAC-RYA/


def solution(A):
    result = 1

    permutation = set(sorted(A))

    if (len(permutation) == len(A)):
        for index, number in enumerate(permutation):
            if (index + 1 != number):
                result = 0
    else:
        result = 0

    return result

File: perm_check/perm_check.js
// https://app.codility.com/demo/results/trainingTT88ZF-JG7/

function solution(A) {
  const len = A.length;
  const set = new Set(A);

  if (set.size !== len) {
    return 0;
  }

  for (let i = 1; i <= len; i++) {
    if (!set.has(i)) {
      return 0;
    }
  }

  return 1;
}
